Data Structures/Array/amortisation.cpp

```diff
@@ -0,0 +1,99 @@
+// Name : Shivam Singhal
+// Email ID : shivamsinghal0610@gmail.com
+ 
+// Array doubling and amortised complexity
+ 
+#include <iostream>
+using namespace std;
+ 
+// function to double array size using amortised technique
+void array_doubling(int c, int arr[], int* size)
+{
+	//doubling array size (using call by reference)
+	*size = (*size) * 2;
+	// creating temporary array of double size
+	int double_arr[*size];
+	// copying existing elements in original array to temporary array
+	for(int i = 0; i < c; i++)
+		double_arr[i] = arr[i];
+	// deleting memory allocated by original array
+	delete [] arr;
+	// re-inserting values into original array with double size
```

Comment:
After deletion of old arr. Have you declared new arr with double size?

Reply:
I didn't get you ?

Reply:
After you have deleted old `arr` with `delete arr`. Where is new `arr`?

Reply:
I have passed the array as a parameter so wont it just update the original array ?